src/Nethermind/Nethermind.TxPool/TxPoolSender.cs

```diff
@@ -1,28 +1,56 @@
 // SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
-// SPDX-License-Identifier: LGPL-3.0-only 
+// SPDX-License-Identifier: LGPL-3.0-only
 
 using System;
 using System.Threading.Tasks;
 using Nethermind.Core;
 using Nethermind.Core.Crypto;
+using Nethermind.Crypto;
 
 namespace Nethermind.TxPool
 {
     public class TxPoolSender : ITxSender
     {
         private readonly ITxPool _txPool;
         private readonly ITxSealer _sealer;
+        private readonly INonceManager _nonceManager;
+        private readonly IEthereumEcdsa _ecdsa;
 
-        public TxPoolSender(ITxPool txPool, ITxSealer sealer)
+        public TxPoolSender(ITxPool txPool, ITxSealer sealer, INonceManager nonceManager, IEthereumEcdsa ecdsa)
         {
             _txPool = txPool ?? throw new ArgumentNullException(nameof(txPool));
             _sealer = sealer ?? throw new ArgumentNullException(nameof(sealer));
+            _nonceManager = nonceManager ?? throw new ArgumentNullException(nameof(nonceManager));
+            _ecdsa = ecdsa ?? throw new ArgumentException(nameof(ecdsa));
         }
 
         public ValueTask<(Keccak, AcceptTxResult?)> SendTransaction(Transaction tx, TxHandlingOptions txHandlingOptions)
         {
+            bool manageNonce = (txHandlingOptions & TxHandlingOptions.ManagedNonce) == TxHandlingOptions.ManagedNonce;
+            tx.SenderAddress ??= _ecdsa.RecoverAddress(tx);
+            if (tx.SenderAddress is null)
+                throw new ArgumentNullException(nameof(tx.SenderAddress));
```

Comment:
Why did you move recovery and throwing outside of `if(managedNonce)`? It was there to avoid heavy operation of `RecoverAddress` if not necessary